internal/resources/machinelearning/resource_alert.go

```diff
@@ -0,0 +1,478 @@
+						path.MatchRelative().AtParent().AtName("outlier_id"),
+					),
+				},
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of the alert.",
+				Computed:    true,
+			},
+			"title": schema.StringAttribute{
+				Description: "The title of the alert.",
+				Required:    true,
+			},
+			"anomaly_condition": schema.StringAttribute{
+				Description: "The condition for when to consider a point as anomalous.",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("any", "low", "high"),
+					stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("outlier_id")),
+				},
+			},
+			"for": schema.StringAttribute{
+				Description: "How long values must be anomalous before firing an alert.",
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString("0s"),
+				Validators: []validator.String{
+					anyDuration(),
+				},
+			},
+			"threshold": schema.StringAttribute{
+				Description: "The threshold of points over the window that need to be anomalous to alert.",
+				Optional:    true,
+			},
+			"window": schema.StringAttribute{
+				Description: "How much time to average values over",
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString("0s"),
+				Validators: []validator.String{
+					maxDuration(12 * time.Hour),
+				},
+			},
+			"labels": schema.MapAttribute{
+				Description: "Labels to add to the alert generated in Grafana.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"annotations": schema.MapAttribute{
+				Description: "Annotations to add to the alert generated in Grafana.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"no_data_state": schema.StringAttribute{
+				Description: "How the alert should be processed when no data is returned by the underlying series",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("Alerting", "NoData", "OK"),
+				},
+			},
+		},
+	}
+}
+
+func (r *alertResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Import ID looks like: /(jobs|outliers)/<jobID>/alerts/<alertID>
+	id := strings.TrimLeft(req.ID, "/")
+	parts := strings.Split(id, "/")
+	if len(parts) != 4 ||
+		(parts[0] != "jobs" && parts[0] != "outliers") ||
+		parts[2] != "alerts" {
+		resp.Diagnostics.AddError("Invalid import ID format", "Import ID must be in the format '/(jobs|outliers)/<jobID>/alerts/<alertID>'")
+		return
+	}
+	model := resourceAlertModel{
+		ID: types.StringValue(parts[3]),
+	}
+	if parts[0] == "jobs" {
+		model.JobID = types.StringValue(parts[1])
+	} else {
+		model.OutlierID = types.StringValue(parts[1])
+	}
+
+	data, diags := r.read(ctx, model)
+	if diags != nil {
+		resp.Diagnostics = diags
+		return
+	}
+	if data == nil {
+		resp.Diagnostics.AddError("Resource not found", "Resource not found")
```

Comment:
maybe `resource` instead of the capital for consistency

Reply:
I changed most of the other ones to consistently use sentence case since this is a user facing error.